addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -1,71 +1,158 @@
-     */
-    init: function (parent, options) {
-        this._super.apply(this, arguments);
-        const editableDocument = options.editable ? options.editable.ownerDocument : document;
+     * @static
+     */
+    static async loadDependencies(getArch) {
+        // Public user using the editor may have a colorpalette but with
+        // the default wysiwyg ones.
+        if (!session.is_website_user) {
+            // We can call the colorPalette multiple times but only need 1 rpc
+            if (!colorpickerTemplateProm && !colorpickerArch) {
+                colorpickerTemplateProm = getArch().then(arch => colorpickerArch = arch);
+
+            }
+            return colorpickerTemplateProm;
+        }
+    }
+    static template = 'web_editor.snippet.option.colorpicker';
+    static props = {
+        getDocument: { type: Function, optional: true },
+        resetTabCount: { type: Number, optional: true },
+        selectedCC: { type: String, optional: true },
+        selectedColor: { type: String, optional: true },
+        resetButton: { type: Boolean, optional: true },
+        excluded: { type: Array, optional: true },
+        excludeSectionOf: { type: Array, optional: true },
+        withCombinations: { type: Boolean, optional: true },
+        noTransparency: { type: Boolean, optional: true },
+        opacity: { type: Number, optional: true },
+        selectedTab: { type: String, optional: true },
+        withGradients: { type: Boolean, optional: true },
+        getArch: { type: Function,  optional: true },
+        onSetColorNames: { type: Function, optional: true },
+        onColorHover: { type: Function, optional: true },
+        onColorPicked: { type: Function, optional: true },
+        onCustomColorPicked: { type: Function, optional: true },
+        onColorLeave: { type: Function, optional: true },
+        onEnterColorpicker: { type: Function, optional: true },
+        getCustomColors: { type: Function, optional: true },
+        getEditableCustomColors: { type: Function, optional: true },
+    };
+    static defaultProps = {
+        getDocument: () => window.document,
+        resetTabCount: 0,
+        resetButton: true,
+        excluded: [],
+        excludeSectionOf: null,
+        withCombinations: false,
+        noTransparency: false,
+        opacity: 1,
+        selectedTab: 'theme-colors',
+        withGradients: false,
+        onSetColorNames: () => {},
+        onColorHover: () => {},
+        onColorPicked: () => {},
+        onCustomColorPicked: () => {},
+        onColorLeave: () => {},
+        onEnterColorpicker: () => {},
+        getCustomColors: () => [],
+        getEditableCustomColors: () => [],
+    }
+    static components = { Colorpicker };
+    elRef = useRef('el');
+    state = useState({
+        showGradientPicker: false,
+    });
+    setup() {
+        this.init();
+        onWillStart(async () => {
+            if (this.props.getArch) {
```

Comment:
It's been a while and I'm replying on the conversation biew where I only see 4 lines of context so maybe I'm just missing something obvious but it is not clear to me why this needs to be a function rather than some kind of reference to a template.

Reply:
Because it is the adaptation of https://github.com/odoo/odoo/blob/ff2c9fc195bba060c3c45e42fb4dd9fc886b782d/addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js#L354C3-L354C3